### PR TITLE
Network specific address decoding

### PR DESCRIPTION
**Pull request by dastanbeksamatov**
## Issue 1

Currently [`indexer`](https://github.com/open-web3-stack/open-web3.js/tree/master/packages/indexer) decodes and stores `SS58` addresses in a generic `Substrate` format, i.e with prefix `5`. I think it would be good enhancement to make it detect the format prefix and decode addresses as belonging to the specific network. `Polkadot` addresses start with a prefix `1`, `Kusama` `G, H, F, etc.` letters.

For more information about address formats, [click here](https://wiki.polkadot.network/docs/learn-accounts#address-format).

This PR will fix this issue by detecting network address prefix and injecting it to registry, so that addresses are stored in a correct format.

## Issue 2

If you try to index remote `Substrate` archive node, both `Scanner` and `Indexer` fails to connect to websocket address with this error:

```bash
Error: WebSocket is not connected
```

The code is:
```js
  ...
  const wsUrl = 'wss://kusama-rpc.polkadot.io';
  const wsProvider = new WsProvider(wsUrl, 100);

  const scanner = new Scanner({
    wsProvider,
  });
  ...
```
 
Strangely this error disappears if you instantiate `ApiPromise` from the `WsProvider` instance. 

```js
...
  const wsUrl = 'wss://kusama-rpc.polkadot.io';
  const wsProvider = new WsProvider(wsUrl, 100);
  const _api = await ApiPromise.create({ provider: wsProvider });
  
  const scanner = new Scanner({
    wsProvider,
  });
...
```

This issue arises only while indexing remote nodes, local sync nodes work fine. 

Changes made in the `indexer.ts` file also fix this issue.

### CODE REVIEW

**Comment by ntduan** [comment]
About issue2. I ran the test case (it connects to `wss://kusama-rpc.polkadot.io`, use the command `npx jest Scanner.test` to run) and there is no `Error: WebSocket is not connected`. 
.

File: packages/scanner/src/Scanner.ts [hunk 1]
```diff
@@ -219,6 +220,11 @@ class Scanner {
       const registry = new TypeRegistry();
       registry.register(this.getSpecTypes(runtimeVersion));
 
+      const chainProperties = registry.createType('ChainProperties', {
+        ss58Format: this.api ? this.api.registry.chainSS58 : 42
```
[the remark this answers is not in the document]
Comment (dastanbeksamatov) [reply packages/scanner/src/Scanner.ts:224]: yes. new instance of `TypeRegistry` has a default `chainSS58` of `42` (generic Substrate address prefix)

[the remark this answers is not in the document]
Comment (ntduan) [reply packages/scanner/src/Scanner.ts:224]: What I mean is that we already have `registry`, so we don't need to access it through polkadot api. Import `apiPromise` is also not necessary.

[the remark this answers is not in the document]
Comment (dastanbeksamatov) [reply packages/scanner/src/Scanner.ts]: Ok. I found a way to get `chainSS58` without using `polkadot api`, sending rpc query instead

**Comment by dastanbeksamatov** [comment]
> About issue2. I ran the test case (it connects to `wss://kusama-rpc.polkadot.io`, use the command `npx jest Scanner.test` to run) and there is no `Error: WebSocket is not connected`.
> .

Test cases run fine. The issue arises when I try to run a scanner or indexer in a NodeJs file.

> About issue2. I ran the test case (it connects to `wss://kusama-rpc.polkadot.io`, use the command `npx jest Scanner.test` to run) and there is no `Error: WebSocket is not connected`.
> .

test passes for me as well, but if you try to run a script using `Scanner` or `Indexer` it won't connect to remote websocket address. See examples [here](https://github.com/dastanbeksamatov/open-web3.js/tree/demo/examples)

**Comment by ntduan** [comment]
> > About issue2. I ran the test case (it connects to `wss://kusama-rpc.polkadot.io`, use the command `npx jest Scanner.test` to run) and there is no `Error: WebSocket is not connected`.
> > .
> 
> test passes for me as well, but if you try to run a script using `Scanner` or `Indexer` it won't connect to remote websocket address. See examples [here](https://github.com/dastanbeksamatov/open-web3.js/tree/demo/examples)

I did not find the error.
```
> node scanner.js
blockN: 4816032 - author: H7Ycyo6X2MQvmNMCFcPbGUiD4bhj3Mhr3oNNmuzHF5d2dmX
```

**Review by ntduan — APPROVED** [review]

**Pull request merged**